src/Microsoft.Net.Http.Headers/DateTimeFormatter.cs

```diff
@@ -0,0 +1,161 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Globalization;
+using static System.Text.Encoding;
+
+namespace Microsoft.Net.Http.Headers
+{
+    internal static class DateTimeFormatter
+    {
+        private static readonly DateTimeFormatInfo FormatInfo = CultureInfo.InvariantCulture.DateTimeFormat;
+
+        private static readonly byte[] MonBytes = UTF8.GetBytes(GetDayName(DayOfWeek.Monday));
+        private static readonly byte[] TueBytes = UTF8.GetBytes(GetDayName(DayOfWeek.Tuesday));
+        private static readonly byte[] WedBytes = UTF8.GetBytes(GetDayName(DayOfWeek.Wednesday));
+        private static readonly byte[] ThuBytes = UTF8.GetBytes(GetDayName(DayOfWeek.Thursday));
+        private static readonly byte[] FriBytes = UTF8.GetBytes(GetDayName(DayOfWeek.Friday));
+        private static readonly byte[] SatBytes = UTF8.GetBytes(GetDayName(DayOfWeek.Saturday));
+        private static readonly byte[] SunBytes = UTF8.GetBytes(GetDayName(DayOfWeek.Sunday));
+
+        private static readonly byte[] JanBytes = UTF8.GetBytes(GetMonthName(1));
+        private static readonly byte[] FebBytes = UTF8.GetBytes(GetMonthName(2));
+        private static readonly byte[] MarBytes = UTF8.GetBytes(GetMonthName(3));
+        private static readonly byte[] AprBytes = UTF8.GetBytes(GetMonthName(4));
+        private static readonly byte[] MayBytes = UTF8.GetBytes(GetMonthName(5));
+        private static readonly byte[] JunBytes = UTF8.GetBytes(GetMonthName(6));
+        private static readonly byte[] JulBytes = UTF8.GetBytes(GetMonthName(7));
+        private static readonly byte[] AugBytes = UTF8.GetBytes(GetMonthName(8));
+        private static readonly byte[] SepBytes = UTF8.GetBytes(GetMonthName(9));
+        private static readonly byte[] OctBytes = UTF8.GetBytes(GetMonthName(10));
+        private static readonly byte[] NovBytes = UTF8.GetBytes(GetMonthName(11));
+        private static readonly byte[] DecBytes = UTF8.GetBytes(GetMonthName(12));
```

Comment:
Uuh. That's a good question. I've been back and forth between `char*` and `byte*` a couple of times because of missing APIs for different targets. I'll revert back to `char[]`.